client/src/components/CreationMode.vue

```diff
@@ -0,0 +1,233 @@
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import { Ref } from '@vue/composition-api';
+
+export interface NewTrackSettings {
+    mode: string;
+    type: string;
+    modeSettings: {
+        Track: {
+            autoAdvanceFrame: boolean;
+        };
+        Detection: {
+            continuous: boolean;
+        };
+    };
+}
+
+export default Vue.extend({
+  name: 'CreationMode',
+  props: {
+    allTypes: {
+      type: Object as PropType<Ref<Array<string>>>,
+      required: true,
+    },
+  },
+  data: () => ({
+    itemHeight: 45, // in pixels
+    help: {
+      mode: {
+        Track: 'Track Mode - advance a frame while drawing',
+        Detection: 'Detection Mode - Used to create multiple detections on a single frame.',
+      },
+      type: 'Choose a default type for the new Track/Detection to be or type in a new type to add it',
+      autoAdvanceFrame: 'After creating a track advance to the next frame.  Hit Esc or do single click to exit.',
+      continuous: 'Immediately stay in detection creation mode after creating a new track.  Hit Esc or do single click to exit.',
+    },
+    modes: ['Track', 'Detection'],
+    selectedMode: 'Track',
+    selectedType: 'unknown',
+    modeSettings: {
+      Track: {
+        autoAdvanceFrame: true,
+      },
+      Detection: {
+        continuous: false,
+      },
+    },
```

Comment:
Keeping this state here and emitting it to the above component probably isn't what we want.  I understand that this is a half-implementation before local storage happens, but even without the persistence, you'll have to accept this state as props and emit it back to the parent.  

I _do_ think it makes sense to put this in a `useCreationSettings.ts`, where you have this reactive state and an update function.  Leaving an reactive copy in `useModeManager` that isn't in sync with this version is sort of confusing.

Reply:
Since this is essentially hierarchical passing up and down of state `Viewer -> SideBar -> TypeList -> CreationMode: CreationMode -> TypeList -> SideBar -> Viewer`, would it make more sense to place this into a vuex module then?

Reply:
You're not wrong. I guess that just depends on whether you want to open the Vuex can of worms yet, because I think the same argument applies to other things we've so far been putting in composition functions.  Everything in `useTrackStore` could also just as easily be a vuex module.

A vuex module + https://www.npmjs.com/package/vuex-persist would be one decent option.

Perhaps we could start with this one thing and see how we like it, then talk about what else should actually be in vuex?